pkg/operators/process/process.go

```diff
@@ -0,0 +1,777 @@
+
+	// Default values
+	defaultInterval = 60 * time.Second
+
+	// Field names
+	fieldPID           = "pid"
+	fieldPPID          = "ppid"
+	fieldCommand       = "command"
+	fieldCPUUsage      = "cpuUsage"
+	fieldMemoryRSS     = "memoryRSS"
+	fieldMemoryVirtual = "memoryVirtual"
+	fieldThreadCount   = "threadCount"
+	fieldState         = "state"
+	fieldUid           = "uid"
+	fieldStartTime     = "startTime"
+	fieldStartTimeStr  = "startTimeStr"
+	fieldMountNsID     = "mountnsid"
+
+	// Clock ticks per second (typically 100 on most Linux systems)
+	// This could be determined dynamically using sysconf(_SC_CLK_TCK) in C
+	// or by reading /proc/stat and calculating based on uptime
+	clockTicksPerSecond = 100
+)
+
+// ProcessInfo represents information about a running process
+type ProcessInfo struct {
+	PID           int       `json:"pid"`
+	PPID          int       `json:"ppid"`
+	Command       string    `json:"command"`
+	CPUUsage      float64   `json:"cpuUsage"`      // CPU usage in percentage
+	MemoryRSS     uint64    `json:"memoryRSS"`     // Resident Set Size in bytes
+	MemoryVirtual uint64    `json:"memoryVirtual"` // Virtual memory size in bytes
+	ThreadCount   int       `json:"threadCount"`   // Number of threads
+	State         string    `json:"state"`         // Process state (R: running, S: sleeping, etc.)
+	Uid           uint32    `json:"uid"`           // UID of the process owner
+	StartTime     uint64    `json:"startTime"`     // Process start time (clock ticks since system boot)
+	StartTimeStr  time.Time `json:"startTimeStr"`  // Process start time as a formatted string
+	MountNsID     uint64    `json:"mountnsid"`     // Mount namespace ID
+}
+
+type processOperator struct{}
+
+func (p *processOperator) Name() string {
+	return Name
+}
+
+func (p *processOperator) Init(globalParams *params.Params) error {
+	return nil
+}
+
+func (p *processOperator) GlobalParams() api.Params {
+	return api.Params{}
+}
+
+func (p *processOperator) InstanceParams() api.Params {
+	return api.Params{}
+}
+
+func (p *processOperator) InstantiateDataOperator(gadgetCtx operators.GadgetContext, instanceParamValues api.ParamValues) (operators.DataOperatorInstance, error) {
+	// Get configuration from viper
+	config, ok := gadgetCtx.GetVar("config")
+	if !ok {
+		return nil, fmt.Errorf("config not found in gadget context")
+	}
+
+	viperConfig, ok := config.(*viper.Viper)
+	if !ok {
+		return nil, fmt.Errorf("config is not a viper instance")
+	}
+
+	// Check if process monitoring is enabled
+	enabled := viperConfig.GetBool(configKeyEnabled)
+	if !enabled {
+		gadgetCtx.Logger().Debug("Process monitoring is disabled")
+		return nil, nil
+	}
+
+	// Get the interval from config or use default
+	interval := viperConfig.GetDuration(configKeyInterval)
+	if interval <= 0 {
+		interval = defaultInterval
+		gadgetCtx.Logger().Debugf("Using default interval: %s", interval)
+	}
+
+	// Get the enabled fields from config
+	enabledFields := make(map[string]bool)
```

Comment:
What about this alternative approach?
1. Do not use this map, base all the logic on the fields variable below
2. The logic to set the default could be handled by something like

```go
if len(fields) == 0 {
  fields = []string{fieldPid, fieldPPID, etc..}
```

3. Add some logic to ensure pid and mountns is always present
4. The logic to create the datasources could be something like

```go
for _, fieldName := range fields {
select {
case fieldPid:
 instance.pidField := ...
case fieldPpid:
...
}
}
```

Reply:
I changed the implementation to your suggested one, thanks!